src/main/java/seedu/address/storage/JsonAdaptedTask.java

```diff
@@ -28,28 +36,23 @@
      * Constructs a {@code JsonAdaptedTask} with the given task details.
      */
     @JsonCreator
-    public JsonAdaptedTask(@JsonProperty("name") String name, @JsonProperty("tag") JsonAdaptedTag tag,
+    public JsonAdaptedTask(@JsonProperty("name") String name, @JsonProperty("tag") List<JsonAdaptedTag> tags,
                               @JsonProperty("priority") String priority,
                            @JsonProperty("date") String date, @JsonProperty("status") String status) {
         this.name = name;
-        // this.tag = tag;
         this.priority = priority;
         this.date = date;
         this.status = status;
+        if (tags != null) {
+            this.tags.addAll(tags);
+        }
     }
 
     /**
      * Converts a given {@code Task} into this class for Jackson use.
      */
     public JsonAdaptedTask(Task source) {
         name = source.getName().get().getValue();
-        /*
-        if (source.getTag() == null) {
-            tag = null;
-        } else {
-            tag = new JsonAdaptedTag(source.getTag().get());
-        }
-        */
         if (source.getPriority().get() == null) {
             priority = null;
         } else {
```

Comment:
I think we should use `isEmpty()` method for the if statement i.e. `source.getPriority().get().isEmpty()`. This is because `get()` method from an empty Optional will not return a `null` but instead throw a `NoSuchElementException`.